### PR TITLE
Fix behaviour of absoluteURI

### PR DESCRIPTION
**Pull request by purplefox**
@cescoffier @pmlopes @vietj please review

Fix for https://bugs.eclipse.org/bugs/show_bug.cgi?id=470125

### CODE REVIEW

File: src/test/java/io/vertx/test/core/HttpTest.java [hunk 1]
```diff
@@ -4082,6 +4082,30 @@ public void start() throws Exception {
...
+    server.requestHandler(req -> {
+      String absURI = req.absoluteURI();
+      assertEquals("http://localhost:8080/path", absURI);
+      req.response().end();
+    });
+    server.listen(onSuccess(s -> {
+      String host = "localhost";
+      String path = "/path";
+      int port = 8080;
+      System.out.println("Client: GET Request-URI = http://" + host + ":" + port + path);
```
Comment (vietj) [inline src/test/java/io/vertx/test/core/HttpTest.java]: println should be removed

**Comment by purplefox** [comment]
If someone could take a quick look that would be great, but if not I'm probably going to have to merge anyway as the next release is coming up fast.

**Comment by cescoffier** [comment]
I gonna have a look this morning.

Sent from my iPhone

On 18 juin 2015, at 08:31, Tim Fox <notifications@github.com> wrote:

If someone could take a quick look that would be great, but if not I'm
probably going to have to merge anyway as the next release is coming up
fast.

—
Reply to this email directly or view it on GitHub
<https://github.com/eclipse/vert.x/pull/1073#issuecomment-113051653>.

**Comment by leolux** [comment]
Thanks! I updated my reproducer because the HTTP Server Metrics still contains 0.0.0.0
https://github.com/leolux/vertx-absolute-uri

**Comment by purplefox** [comment]
I think the metrics naming is a separate issue, and is due to the way the metrics name is identified is using host and port - but of course this is not unique.

In Vert.x you can have many servers listening on port 0.0.0.0, and the port value it uses is not the actual port.

@vietj

**Comment by purplefox** [comment]
I've made a small change in core so different http server metrics with same host/port are now distinguishable.

But the server metric will still say 0.0.0.0 if you told your server to listen on port 0.0.0.0. At metrics creation time it's not possible to know the value of Server header sent to the server from the client.

**Pull request merged**